src/routing/tests/mod.rs

```diff
@@ -390,6 +340,136 @@ async fn handle_consensus_on_online_of_elder_candidate() -> Result<()> {
+    let mut relocate_sent = false;
+    let mut relocate_age = 0;
+    for command in commands {
+        let (message, recipients) = match command {
+            Command::SendMessage {
+                recipients,
+                message,
+                ..
+            } => (Message::from_bytes(&message).unwrap(), recipients),
+            _ => continue,
+        };
+
+        match message.variant() {
+            Variant::NodeApproval(proven_elders_info) => {
+                assert_eq!(proven_elders_info.value, elders_info);
+                assert_eq!(recipients, [*peer.addr()]);
+                node_approval_sent = true;
+            }
+            Variant::Vote { content, .. } => {
+                let message = if let Vote::SendMessage { message, .. } = content {
+                    message
+                } else {
+                    continue;
+                };
+
+                let details = if let Variant::Relocate(details) = &message.variant {
+                    details
+                } else {
+                    continue;
+                };
+
+                assert_eq!(details.pub_id, *peer.name());
+                assert_eq!(details.destination, *peer.name());
+                relocate_age = details.age;
+                assert_eq!(recipients, [*peer.addr()]);
+                relocate_sent = true;
+            }
+            _ => continue,
+        }
+    }
+
+    Ok((node_approval_sent, relocate_sent, relocate_age))
+}
+
+async fn handle_consensus_on_online_of_rejoined_node(phase: NetworkPhase, age: u8) -> Result<()> {
+    let prefix = match phase {
+        NetworkPhase::Startup => Prefix::default(),
+        NetworkPhase::Regular => "0".parse().unwrap(),
+    };
+    let (elders_info, mut nodes) = gen_elders_info(prefix, ELDER_SIZE);
+    let sk_set = SecretKeySet::random();
+    let (mut section, section_key_share) = create_section(&sk_set, &elders_info)?;
+
+    // Make a left peer.
+    let peer = create_peer().with_age(age);
+    let member_info = MemberInfo {
+        peer,
+        state: PeerState::Left,
+    };
+    let member_info = proven(sk_set.secret_key(), member_info)?;
+    let _ = section.update_member(member_info);
+
+    // Make a Node
+    let (event_tx, mut event_rx) = mpsc::unbounded_channel();
+    let node = nodes.remove(0);
+    let state = Approved::new(node, section, Some(section_key_share), event_tx);
+    let stage = Stage::new(state, create_comm()?);
+
+    // Simulate peer with the same name is rejoin and verify resulted behaviours.
+    let (node_approval_sent, relocate_sent, relocate_age) =
+        handle_online_command(&peer, &sk_set, &stage, elders_info).await?;
+    assert!(event_rx.try_recv().is_err());
+    // After startup, a rejoin node with low age will be rejected.
+    if let NetworkPhase::Regular = phase {
+        if age / 2 <= MIN_AGE {
+            assert!(!node_approval_sent);
+            assert!(!relocate_sent);
+            assert!(relocate_age == 0);
```

Comment:
Sorry for nitpicking, but this looks a bit weird. How about instead of having separate `relocate_sent` and `relocate_age`, have only `relocate_age: Option<u8>` where `Some` means the `Relocate` message was sent with the given age and `None` means no `Relocate` message was sent?